tests/functional/test_integration.py

```diff
@@ -0,0 +1,12 @@
+import pytest
+from ops.model import ActiveStatus
+from pytest_operator.plugin import OpsTest
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    charm = await ops_test.build_charm(".")
+    resources = {
+        "gunicorn-image": "localhost:32000/gunicorn:latest"
+    }
```

Comment:
I'm not sure this is needed is it? Take a look at https://github.com/canonical/indico-operator/blob/main/tests/integration/conftest.py and https://github.com/canonical/indico-operator/blob/main/tests/integration/test_charm.py. The indico charm has one external image that it's using (for nginx-prometheus-exporter) but the other image (indico-image, indico-nginx-image) are built as part of the test, and that's the pattern we should be copying here.